File: frontend/src/types/product.ts
// 상품 타입. 백엔드 schemas/product.py 와 동기화.

export type ProductStatus = "DRAFT" | "ACTIVE" | "SOLD_OUT" | "ARCHIVED" | "LOW_STOCK";

export interface Product {
  id: number;
  name: string;
  category: string;
  price: number;
  stock: number;
  status: ProductStatus;
}

export interface ProductListResponse {
  items: Product[];
  total: number;
  page: number;
  size: number;
}

File: backend/app/schemas/product.py
"""상품 API 스키마."""

from pydantic import BaseModel, ConfigDict, Field

from app.models.product import ProductStatus


class ProductBase(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    category: str = Field(min_length=1, max_length=50)
    price: int = Field(ge=0)
    stock: int = Field(default=0, ge=0)
    brand_id: int | None = None


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=200)
    category: str | None = Field(default=None, max_length=50)
    price: int | None = Field(default=None, ge=0)
    stock: int | None = Field(default=None, ge=0)
    status: ProductStatus | None = None
    brand_id: int | None = None


class ProductOut(ProductBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    status: ProductStatus


class ProductListOut(BaseModel):
    items: list[ProductOut]
    total: int
    page: int
    size: int